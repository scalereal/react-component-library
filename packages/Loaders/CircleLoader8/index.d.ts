import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
declare const CircleLoader8: React.FC<Props> & WithStyle;
export default CircleLoader8;
