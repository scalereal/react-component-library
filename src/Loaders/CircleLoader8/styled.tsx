import styled from 'styled-components';
import { Props } from '../types';

export const Loader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    color: ${pr=>pr.Color?pr.Color:'inherit'};
    vertical-align: middle;
    pointer-events: none;
    
    position: relative;
	animation: loader-10-1 2.0s infinite linear;
	&:before,
	&:after {
		content: '';
		width: 0;
		height: 0;
		border: .5em solid currentcolor;
		display: block;
		position: absolute;
		border-radius: 100%;
		animation: loader-10-2 2s infinite ease-in-out;
	}
	&:before {
		top: 0;
		left: 50%;
	}
	&:after {
		bottom: 0;
		right: 50%;
		animation-delay: -1s;
	}

    @keyframes loader-10-1 {
        100% {
            transform: rotate(360deg);
        }
    }

    @keyframes loader-10-2 {
        0%,
        100% {
            transform: scale(0);
        }
        50% {
            transform: scale(1);
        }
    }
`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
            ${Loader}{
                margin: 5px 0 5px 10px;
            }
           `;
        case "XS":
            return `font-size: 1rem;
            ${Loader}{
                margin: 7px 0 7px 10px;
            }
            `;
        case "S":
            return `font-size: 2rem;
            ${Loader}{
                margin: 15px 0 15px 20px;
            }
            `;
        
        case "M":
            return `font-size: 3rem;
            ${Loader}{
                margin: 18px 0 18px 25px;
            }
            `;

        case "L":
            return `font-size: 5rem;
            ${Loader}{
                margin: 28px 0 28px 40px;
            }
            `;
        
        case "XL":
            return `font-size: 6rem;
            ${Loader}{
                margin: 35px 0 35px 50px;
            }
            `;
  
      default:
        return `font-size: 1rem;
        ${Loader}{
            margin: 15px 0 15px 20px;
        }
       `;
    }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.display?'inline-block':'none'};
`;
