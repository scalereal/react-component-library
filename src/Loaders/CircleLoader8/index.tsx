import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

const CircleLoader8:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

CircleLoader8.displayName="CircleLoader8";
CircleLoader8.defaultProps = {
    size:'S',
    display: true
};

export default CircleLoader8;
