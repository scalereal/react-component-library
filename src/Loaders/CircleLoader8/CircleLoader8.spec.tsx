import { render } from '@testing-library/react';
import * as React from 'react';
import CircleLoader8 from '.';

describe('CircleLoader8 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<    CircleLoader8/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <    CircleLoader8
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <    CircleLoader8 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});