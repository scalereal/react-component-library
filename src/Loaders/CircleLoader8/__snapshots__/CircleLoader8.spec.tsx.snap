// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleLoader8 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz bozTVO"
  >
    <div
      class="sc-bdfBwQ hrZhnP"
    />
  </div>
</div>
`;

exports[`CircleLoader8 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz TLVsP"
  >
    <div
      class="sc-bdfBwQ dNRfHe"
    />
  </div>
</div>
`;

exports[`CircleLoader8 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz gQcywn"
  >
    <div
      class="sc-bdfBwQ hrZhnP"
    />
  </div>
</div>
`;
